cpmonitor/models.py

```diff
@@ -295,17 +296,27 @@ def __str__(self) -> str:
         return self.title
 
     slugs = models.SlugField(
-        "slugs",
+        "In der URL",
         max_length=255,
         editable=False,
     )
 
-    def clean(self):
-        """Set / update the slugs on every validation. (Done by admin before `save()`)"""
-        self.slugs = ""
-        for anc in self.get_ancestors():
-            self.slugs += slugify(anc.title) + "/"
-        self.slugs += slugify(self.title)
+    @staticmethod
+    def _get_slugs_with_parent(new_parent, title):
+        """Determine the `slugs` field based on the new parents slugs and title."""
+        if new_parent:
+            return new_parent.slugs + "/" + slugify(title)
+        else:
+            return slugify(title)
+
+    @staticmethod
+    def get_slugs_for_move(ref_node, pos, title):
+        """Determine the `slugs` field given treebeards reference data."""
+        if ref_node and isinstance(pos, str) and not pos.endswith("-child"):
+            new_parent = ref_node.get_parent()
+        else:
+            new_parent = ref_node
```

Comment:
I don't fully understand the purpose of the if-condition; in which cases do we use the parent of the ref_node..? Could we add a variable for that to give it a name?
```suggestion
        is_move_to_inner_node_pos(??) = ref_node and isinstance(pos, str) and not pos.endswith("-child")
        if is_move_to_inner_node_pos:
            new_parent = ref_node.get_parent()
        else:
            new_parent = ref_node
```